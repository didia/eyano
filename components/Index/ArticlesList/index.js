// Vendor
import T from 'prop-types';

const ArticleShape = T.shape({
  excerpt: T.string.isRequired,
  image: T.string.isRequired,
  slug: T.string.isRequired,
  title: T.string.isRequired
}).isRequired;

const ArticleItem = ({ article }) => (
  <a className="article" href={`#${article.slug}`}>
    <div className="article__details">
      <h2 className="article__title">{article.title}</h2>

      <p className="article__excerpt">{article.excerpt}</p>
    </div>

    <div className="article__image" />

    <style jsx>
      {`
    .article {
      display: flex;
      min-height: 140px;
      padding: 16px;
      box-shadow: 0 2px 2px rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.2), 0 1px 5px rgba(0,0,0,.12);
      text-decoration: none;
      transition: 0.3s box-shadow ease-in-out;
    }

    .article:hover,
    .article:active {
      box-shadow: 0 0 8px rgba(0,0,0,.18), 0 8px 16px rgba(0,0,0,.36);
    }

    .article__image {
      flex: 0 0 33%;
      margin-left: 16px;
      background: no-repeat url("${article.image}");
      background-size: cover;
      background-position: center;
    }

    .article__title {
      font-size: 24px;
      font-weight: 700;
      line-height: 1.3;
      color: #006b6b;
      transition: 0.2s color ease-in-out;
    }

    .article:hover .article__title,
    .article:active .article__title {
      color: #26baba;
    }


    .article__excerpt {
      margin-top: 16px;
      line-height: 1.5;
      color: #6e767d;
    }
  `}
    </style>
  </a>
);

ArticleItem.propTypes = {
  article: ArticleShape
};

const ArticlesList = ({ articles }) => {
  return (
    <ul>
      {articles.map((article, index) => (
        <li key={`article-${index}`} className="item">
          <ArticleItem article={article} />
        </li>
      ))}

      <style jsx>
        {`
          .item {
            margin-bottom: 16px;
          }
        `}
      </style>
    </ul>
  );
};

ArticlesList.propTypes = {
  articles: T.arrayOf(ArticleShape).isRequired
};

export default ArticlesList;
