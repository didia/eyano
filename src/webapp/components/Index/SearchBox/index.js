// Vendor
import React from 'react';
import { injectIntl, intlShape } from 'react-intl';
import SVG from 'react-inlinesvg';

// Assets
import searchIcon from '../../../static/images/icon-magnifying-glass.svg';
import resetIcon from '../../../static/images/icon-close.svg';

class SearchBox extends React.Component {
  constructor() {
    super();

    this.state = {
      searchText: ''
    };

    this.handleSearchTextChange = this.handleSearchTextChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleReset = this.handleReset.bind(this);
  }

  handleReset(event) {
    this.setState({
      ...this.state,
      searchText: ''
    });

    event.preventDefault();
  }

  handleSearchTextChange(event) {
    this.setState({
      ...this.state,
      searchText: event.target.value
    });
  }

  handleSubmit(event) {
    window.console.log(`Search query is: ${this.state.searchText}`);
    event.preventDefault();
  }

  hasInput() {
    return this.state.searchText.trim() !== '';
  }

  computeFormClassName() {
    const formBaseClass = 'form';
    return this.hasInput() ? `${formBaseClass} form--dirty` : formBaseClass;
  }

  render() {
    const placeHolderText = this.props.intl.formatMessage({
      id: 'components.search-box.placeholder'
    });

    return (
      <form
        className={this.computeFormClassName()}
        onSubmit={this.handleSubmit}
        onReset={this.handleReset}
      >
        <div className="input-wrapper">
          <input
            autoFocus={true}
            className="input"
            placeholder={placeHolderText}
            value={this.state.searchText}
            onChange={this.handleSearchTextChange}
          />

          <button className="reset-button" type="reset">
            <SVG
              className="button-wrapper reset-icon-wrapper"
              src={resetIcon}
              wrapper={React.createFactory('div')}
            />
          </button>
        </div>

        <button
          className="search-button"
          type="submit"
          disabled={!this.hasInput()}
        >
          <SVG
            className="button-wrapper search-icon-wrapper"
            src={searchIcon}
            wrapper={React.createFactory('div')}
          />
        </button>

        <style jsx>{`
          .form {
            display: flex;
            align-items: center;
            box-shadow: 0 16px 32px rgba(0, 44, 80, 0.2);;
          }

          .form :global(.button-wrapper) {
            display: flex;
            align-items: center;
          }

          .input-wrapper {
            display: flex;
            align-items: center;
            width: 100%;
            background-color: #fff;
            padding: 16px;
          }

          .input {
            width: 100%;
            outline: none;
            border: none;
            font-size: 20px
            color: #595959;
            -webkit-font-smoothing: antialiased
          }

          .reset-button {
            overflow: hidden;
            margin-right: 4px;
            opacity: 0;
            transition: 0.3s opacity ease-in-out;
            cursor: pointer;
          }

          .reset-button :global(svg) {
            width: 15px;
            height: 15px;
            fill: #555c5c;
          }

          .search-button {
            height: 57px;
            width: 57px;
            flex-shrink: 0;
            cursor: pointer;
            background-color: #006b6b;
            transition: 0.3s background-color ease-in-out;
          }

          .search-button :global(svg) {
            width: 20px;
            height: 20px;
            fill: #fff;
          }

          .search-button:hover,
          .search-button:active {
            background-color: #4cc7c7;
          }

          .search-button:disabled {
            cursor: default;
            background-color: #008989;
          }

          .search-button :global(.search-icon-wrapper) {
            justify-content: center;
          }

          .form--dirty .reset-button {
            opacity: 1;
          }
        `}</style>
      </form>
    );
  }
}

SearchBox.propTypes = {
  intl: intlShape.isRequired
};

export default injectIntl(SearchBox);
