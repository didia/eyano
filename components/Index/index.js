// Vendor
import { FormattedMessage } from 'react-intl';

import PageWrapper from '../PageWrapper';
import SearchBox from '../SearchBox';

const TopSection = () => (
  <header className="header">
    <PageWrapper>
      <div className="content-wrapper">
        <FormattedMessage id="components.search-box.title">
          {title => <h1 className="title">{title}</h1>}
        </FormattedMessage>
        <div className="search-box">
          <SearchBox />
        </div>
      </div>
    </PageWrapper>
    <style jsx>
      {`
        .header {
          background-color: #00b4b3;
        }

        .content-wrapper {
          padding: 32px 0;
        }

        .title {
          color: #fff;
          text-align: center;
          font-weight: normal;
        }

        .search-box {
          max-width: 600px;
          margin: 0 auto;
        }
      `}
    </style>
  </header>
);

const Index = () => (
  <div>
    <TopSection />
  </div>
);

export default Index;
