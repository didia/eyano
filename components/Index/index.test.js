import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Index from './index';

describe('Index Component', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();

    const component = renderer.render(<Index />);

    expect(component).toMatchSnapshot();
  });
});
