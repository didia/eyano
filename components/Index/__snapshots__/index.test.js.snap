// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Component renders as expected 1`] = `
<div>
  <TopSection />
</div>
`;
